### PR TITLE
Adds Energydrink

### PR DESCRIPTION
**Pull request by ghost**
Recipe: 3 energy drink = 2 teslium + 1 cola + 1 coffee

Addiction threshold: 6u
Overdose: 10u

When uses in proper amounts, it regenerates stamina at a slightly lesser rate than meth.  However, when addicted, you will jitter, become dizzy, randomly scream, and later, move randomly.

When overdosed, it's fatal. The higher dose you get, the more likely you are to die quickly. You will get anywhere from 40-65 OXY damage when overdosed, and sometimes 5-17 TOX damage. The rate at which you get this goes up the higher above the overdose threshold you go.

:cl: Thefastfoodguy
add: Energy drink!
/:cl:

### CODE REVIEW

File: hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 1]
```diff
@@ -232,4 +232,79 @@
...
+	name = "Energy Drink"
+	id = "energydrink"
+	description = "A highly concentrated energy drink."
+	taste_description = "pure energy"
+	color = "#e7f920" // rgb: 231, 249, 32
+	reagent_state = LIQUID
+	overdose_threshold = 11
+	addiction_threshold = 6
+	metabolization_rate = 0.75 * REAGENTS_METABOLISM
+
```
Comment (KayeArray) [inline hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm:248]: weird metab rate to use if you ask me

File: hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 2]
```diff
@@ -232,4 +232,79 @@
...
+		to_chat(M, "<span class='notice'>[mesg]</span>")
+	M.AdjustStun(-30, 0)
+	M.AdjustKnockdown(-30, 0)
+	M.AdjustUnconscious(-30, 0)
+	M.adjustStaminaLoss(-2, 0)
+	M.status_flags |= GOTTAGOREALLYFAST
+	M.Jitter(2)
+	..()
+	. = 1
+
```
Comment (KayeArray) [inline hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm:261]: is this even needed?

Comment (ghost) [reply hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm:261]: copy/pasted from meth

File: hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm (hunk 1 above)
Comment (ghost) [reply hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm:248]: copy/pasted from meth

File: hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 3]
```diff
@@ -232,4 +232,79 @@
...
+/datum/reagent/drug/energydrink/overdose_process(mob/living/M)
+	if(prob(2*(volume/11)))
+		var/painmesg = pick("Your chest hurts alot!", "You feel like you're about to collapse...", "The... pain... hurts!")
+		M.visible_message("<span class='danger'>[M] clutches their chest in pain!</span>", "<span class='userdanger'>[painmesg]</span>")
+		M.adjustOxyLoss(rand(40,65))
```
Comment (KayeArray) [inline hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm]: you might as well just give them a heart attack

Comment (ghost) [reply hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm]: Wait you can do that? how

File: hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 4]
```diff
@@ -232,4 +232,79 @@
...
+
+/datum/reagent/drug/energydrink/addiction_act_stage4(mob/living/carbon/human/M)
+	if(M.canmove && !ismovableatom(M.loc))
+		for(var/i = 0, i < 8, i++)
+			step(M, pick(GLOB.cardinals))
+	M.Jitter(20)
+	M.Dizzy(20)
+	M.adjustToxLoss(5, 0)
+	if(prob(50))
+		M.emote(pick("twitch","drool","moan"))
```
Comment (KayeArray) [inline hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm:311]: message spam

Comment (ghost) [reply hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm:311]: meth does this too

File: hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 5]
```diff
@@ -232,4 +232,79 @@
...
+
+/datum/reagent/drug/energydrink/overdose_process(mob/living/M)
+	if(prob(2*(volume/11)))
+		var/painmesg = pick("Your chest hurts alot!", "You feel like you're about to collapse...", "The... pain... hurts!")
+		M.visible_message("<span class='danger'>[M] clutches their chest in pain!</span>", "<span class='userdanger'>[painmesg]</span>")
+		M.adjustOxyLoss(rand(40,65))
+		if(prob(25))
+			M.adjustToxLoss(rand(5,17))
+	..()
+	. = 1
```
Comment (ghost) [inline hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm]: . = 1 should only run if your code did something special

Comment (ghost) [reply hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm]: >see meth code

File: hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm (hunk 3 above)
Comment (KayeArray) [reply hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm]: if(ishuman(M))
					var/mob/living/carbon/human/H = M
					if(!H.undergoing_cardiac_arrest() && H.can_heartattack())
						H.set_heartattack(TRUE)
						if(H.stat == CONSCIOUS)
							H.visible_message("<span class='userdanger'>[H] clutches at [H.p_their()] chest as if [H.p_their()] heart stopped!</span>")

Comment (ghost) [reply hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm]: if you're clutching at your chest, it should probably apply a stun

File: hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 6]
```diff
@@ -232,4 +232,79 @@
...
+	id = "energydrink"
+	description = "A highly concentrated energy drink."
+	taste_description = "pure energy"
+	color = "#e7f920" // rgb: 231, 249, 32
+	reagent_state = LIQUID
+	overdose_threshold = 11
+	addiction_threshold = 6
+	metabolization_rate = 0.75 * REAGENTS_METABOLISM
+
+/datum/reagent/drug/energydrink/on_mob_life(mob/living/M)
```
Comment (ghost) [inline hippiestation/code/modules/reagents/chemistry/reagents/drug_reagents.dm:249]: this is really broken imo

anti stuns are a bad design and make it difficult to kill people

**Pull request closed without merging**